src/parser.ts

```diff
@@ -155,7 +156,14 @@ function handleToken(
   if (attributes && ['tag', 'fence', 'image'].includes(typeName))
     annotate(node, attributes);
 
-  parent.push(node);
+  if (
+    handleSlots &&
+    tag === 'slot' &&
+    typeof node.attributes.primary === 'string'
+  )
+    parent.slots[node.attributes.primary] = node;
```

Comment:
Should we make `children` a reserved slot key? Or is there no reason to avoid this?

Reply:
I don't think this is necessary. In the event that someone uses `children` as the name of the slot, the value will appear as expected in the render tree but will be replaced by the actual children in the React props.

This is already the behavior we have today with attributes and we don't enforce any restrictions to prevent that. React itself does pretty much the same thing if use a "children" attribute in JSX as well as passing in actual children.